api/sys_auth.go

```diff
@@ -82,10 +82,15 @@ func (c *Sys) DisableAuth(path string) error {
 // documentation. Please refer to that documentation for more details.
 
 type EnableAuthOptions struct {
-	Type        string `json:"type" structs:"type"`
-	Description string `json:"description" structs:"description"`
-	Local       bool   `json:"local" structs:"local"`
-	PluginName  string `json:"plugin_name,omitempty" structs:"plugin_name,omitempty" mapstructure:"plugin_name"`
+	Type        string          `json:"type" structs:"type"`
+	Description string          `json:"description" structs:"description"`
+	Config      AuthConfigInput `json:"config" structs:"config"`
+	Local       bool            `json:"local" structs:"local"`
+	PluginName  string          `json:"plugin_name,omitempty" structs:"plugin_name,omitempty"`
+}
+
+type AuthConfigInput struct {
```

Comment:
There are arguments against it (mostly being, maybe what we need will diverge in the future) but it seems like simply reusing MountConfigInput could be good here. This allows the same structure to be used for tuning both (ttl, maxttl, updated plugin name, etc.).

Reply:
Is `ForceNoCache` a property of auth mounts as well? Can the same be said for `MountConfigOutput` and `AuthConfigOutput`

Reply:
It is. Most mounts do nothing with it, but it is technically able to be set. So -- yes.

Reply:
How about the second question, should we use `MountConfigOutput` for both cases then since the config params should be the same?

Reply:
The problem is that this would be an API change. But -- then we're inconsistent because you get an AuthConfigOutput with a MountConfigInput whereas for secrets you get MountConfigInput/MountConfigOutput.

Part of me says "this is a silly situation to be in already and let's just break it and be done with it" but...we probably shouldn't. So I guess, leave AuthConfigInput after all.